Python/ceval.c

```diff
@@ -4566,10 +4566,16 @@ call_function(PyObject ***pp_stack, Py_ssize_t oparg, PyObject *kwnames)
     }
     else if (Py_TYPE(func) == &PyMethodDescr_Type) {
         PyThreadState *tstate = PyThreadState_GET();
-        if (tstate->use_tracing && tstate->c_profilefunc) {
-            // We need to create PyCFunctionObject for tracing.
-            PyMethodDescrObject *descr = (PyMethodDescrObject*)func;
-            func = PyCFunction_NewEx(descr->d_method, stack[0], NULL);
+        if (nargs > 0 && tstate->use_tracing) {
```

Comment:
Why the check for `tstate->c_profilefunc` is missed?

Reply:
That check is actually redundant. The old code was checking `tstate->use_tracing && tstate->c_profilefunc` twice (once here and once in the `C_TRACE` macro). It's not wrong to keep the additional `tstate->c_profilefunc` check but it's just not needed.